src/cluster.c

```diff
@@ -5409,6 +5409,11 @@ NULL
         }
     } else if (!strcasecmp(c->argv[1]->ptr,"forget") && c->argc == 3) {
         /* CLUSTER FORGET <NODE ID> */
+        if (sdslen(c->argv[2]->ptr) != CLUSTER_NAMELEN) {
+            addReplyError(c, "Invalid length for node id");
+            return;
+        }
```

Comment:
There are multiple places we do this check, like `CLUSTER REPLICATE <NODE ID>`. We should maybe write a small wrapper that does all this validation and put it everywhere.

Reply:
@madolson I'm glad to receive your reply. I checked some codes and found that there are still some places that need to be verified. I'll try to modify them.